### PR TITLE
Fix how scripted parallel stages are displayed.

### LINKED ISSUE #92: Nested build stages don't nest properly in console
### Jenkins and plugins versions report

-

### What Operating System are you using (both controller, and any agents involved in the problem)?

-

### Reproduction steps

see
https://ci.jenkins.io/job/Plugins/job/slack-plugin/job/master/325/pipeline-console/?selected-node=25

it uses https://github.com/jenkins-infra/pipeline-library/blob/master/vars/buildPlugin.groovy

I'll get a minimal pipeline up later on that replicates the same thing

The steps should be nested under the build stage but instead they are nested under linux-11

image

### Expected Results

steps nested under build

### Actual Results

steps nested under linux-11

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by timbrown5**
Fixes #92 
- Moved the logic to add steps to child stages of parallel branches to chunkStart.

Before:
Screenshot 2023-02-20 at 10 19 53

After:
Screenshot 2023-02-20 at 13 48 36

Tested with #210 and it seems to be compatible:
Before:
Screenshot 2023-02-20 at 10 16 14

After: 
Screenshot 2023-02-20 at 13 41 00

Tested with Pipeline:
```
node {
	stage("Build all") {
		parallel "linux-8": {
			retry(3) {
				timeout(time: 60, unit: 'SECONDS') {
					stage ("Checkout (linux-8)") {
						echo "Checking out linux-8"
					}
					stage ("Build (linux-8)") {
						echo "Building linux-8"
					}
					stage ("Archive (linux-8)") {
						echo "Archiving linux-8"
					}
				}
			}
		},
		"linux-11": {
			retry(3) {
				timeout(time: 60, unit: 'SECONDS') {
					stage ("Checkout (linux-11)") {
						echo "Checking out linux-8"
					}
					stage ("Build (linux-11)") {
						echo "Building linux-8"
					}
					stage ("Archive (linux-11)") {
						echo "Archiving linux-8"
					}
				}
			}
		}
	}
	stage ("Deploy") {
		echo "Deploying..."
	}
}
```

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

File: src/main/java/io/jenkins/plugins/pipelinegraphview/utils/PipelineStepVisitor.java [hunk 1]
```diff
@@ -95,23 +95,25 @@ public PipelineStepVisitor(WorkflowRun run, @Nullable final FlowNode node) {
     }
   }
 
-  @Override
-  public void parallelBranchStart(
+  //@Override
+  public void xxxparallelBranchStart(
```
Comment (timja) [inline src/main/java/io/jenkins/plugins/pipelinegraphview/utils/PipelineStepVisitor.java:99]: not used anymore?

**Review by timja — COMMENTED** [review]
spotless is failing

File: src/main/java/io/jenkins/plugins/pipelinegraphview/utils/PipelineStepVisitor.java (hunk 1 above)
Comment (timbrown5) [reply src/main/java/io/jenkins/plugins/pipelinegraphview/utils/PipelineStepVisitor.java:99]: Sorry, work still needed to get all the use-cases working (I thought I could remove this, but it break other things).

**Comment by timbrown5** [comment]
Turned out it was an issue with declarative vs script parallel blocks.

Found I needed to use `isDeclarative` to allow both of these to display properly:
```
pipeline {
    agent any
    stages {
        stage('Parallel Stage') {
            failFast true
            parallel {
                stage('Branch A') {
                    steps {
                        echo "On Branch A - 1"
                        echo "On Branch A - 2"
                    }
                }
                stage('Branch B') {
                    steps {
                        echo "On Branch B - 1"
                        echo "On Branch B - 2"
                    }
                }
            }
        }
    }
}
```
vs
```
node {
	stage("Build all") {
		parallel "linux-8": {
			stage ("Checkout (linux-8)") {
				echo "Checking out linux-8"
			}
			stage ("Build (linux-8)") {
				echo "Building linux-8"
			}
			stage ("Archive (linux-8)") {
				echo "Archiving linux-8"
			}
		},
		"linux-11": {
			stage ("Checkout (linux-11)") {
				echo "Checking out linux-11"
			}
			stage ("Build (linux-11)") {
				echo "Building linux-11"
			}
			stage ("Archive (linux-11)") {
				echo "Archiving linux-11"
			}
		}
	}
	stage ("Deploy") {
		echo "Deploying..."
	}
}
```
(Regression test added).

File: src/main/frontend/pipeline-console-view/pipeline-console/main/DataTreeView.tsx [hunk 2]
```diff
@@ -51,6 +51,7 @@ const getTreeItemsFromStage = (stageItems: StageInfo[], steps: StepInfo[]) => {
   // Copy steps so we don't affect props.steps.
   let stepsCopy = [...steps];
   return stageItems.map((stageItemData) => {
+    console.info(`Generating stage item(s) for '${stageItemData.name} - ${stageItemData.id}'.`)
```
Comment (timja) [inline src/main/frontend/pipeline-console-view/pipeline-console/main/DataTreeView.tsx:54]:
```suggestion
    console.debug(`Generating stage item(s) for '${stageItemData.name} - ${stageItemData.id}'.`)
```

**Review by timja — APPROVED** [review]
Thanks!

**Pull request merged**